src/org/thoughtcrime/securesms/ConversationActivity.java

```diff
@@ -1653,4 +1656,41 @@ public boolean onQueryTextChange(String query) {
     }
     return true; // action handled by listener
   }
+
+  public void initializeContactRequest() {
+    if (!dcChat.isContactRequest()) {
+      messageRequestBottomView.setVisibility(View.GONE);
+      return;
+    }
+
+    messageRequestBottomView.setVisibility(View.VISIBLE);
+    messageRequestBottomView.setAcceptOnClickListener(v -> {
+      dcContext.acceptChat(chatId);
+      messageRequestBottomView.setVisibility(View.GONE);
+      composePanel.setVisibility(View.VISIBLE);
+    });
+    messageRequestBottomView.setBlockOnClickListener(v -> {
+      dcContext.blockChat(chatId);
+      Bundle extras = new Bundle();
+      extras.putInt(ConversationListFragment.RELOAD_LIST, 1);
+      handleReturnToConversationList(extras);
+    });
+
+    String question;
+    if (dcChat.isMailingList()) {
+      question = getString(R.string.ask_show_mailing_list, dcChat.getName());
+    } else {
+      final int[] members = dcContext.getChatContacts(chatId);
+      if (members.length == 1) {
```

Comment:
doesn't this means that if someone creates a group with two members, send a message and then leaves the group, group will have only 1 member and the contact request will be treated as a 1:1 chat with block option, didn't try it yet but seems it is what will happen